src/main/java/com/smoothstack/lms/adminmicroservice/controller/AdminController.java

```diff
@@ -34,6 +40,22 @@
 
 	@Autowired
 	AdminService adminService;
+	
+	@Autowired
+	private DiscoveryClient discoveryClient;
+
+	@RequestMapping("/service-instances/{applicationName}")
+	public List<ServiceInstance> serviceInstancesByApplicationName(@PathVariable String applicationName) {
+		return this.discoveryClient.getInstances(applicationName);
+	}
+
+	@Autowired
+	Environment environment;
+
+	@GetMapping(path = "/administrator/port")
+	public String port() {
+		return environment.getProperty("local.server.port");
+	}
```

Comment:
Your controller isn't the right place for this kind of code. This should go in the main class or in a configuration file.